### ISSUE TITLE
sql: timestamps are truncated to precision 0 after 19.2 upgrade

### ISSUE THREAD

**Issue by lucy-zhang** [cockroachdb/cockroach#42283]
Start a cluster running 19.1.5 and create a table with a timestamp column:
```
root@localhost:26257/defaultdb> create table t (a timestamp);
CREATE TABLE
```
Insert a value (just for comparison's sake):
```
root@localhost:26257/defaultdb> insert into t values ('2016-01-25 10:10:10.555555');
INSERT 1
```
Then upgrade to 19.2.0-rc.2.

Now when timestamps are inserted, they get truncated:
```
root@localhost:26257/defaultdb> insert into t values ('2016-01-25 10:10:10.555555');
INSERT 1
```
```
root@localhost:26257/defaultdb> select * from t;
                 a
+----------------------------------+
  2016-01-25 10:10:10.555555+00:00
  2016-01-25 10:10:11+00:00
(2 rows)
```

This was introduced in #37920. The problem is that the `Precision` field for the timestamp types were unset, and the precision was always 6 (i.e. microseconds) by default. After the change, a `Precision` of 0 is now taken to mean precision 0 (i.e. seconds).

**Comment by RaduBerinde** [comment]
Cool find! I filed #42309 to track adding a test along these lines.

**Comment by solongordon** [comment]
Fixed by #42286.
